tests/integration/test_integration.py

```diff
@@ -523,11 +578,21 @@ def test_synchronous_writer_shutdown_raises_no_exception():
     tracer.shutdown()
 
 
-@parametrize_with_all_encodings
 @skip_if_testagent
-def test_writer_flush_queue_generates_debug_log(caplog, encoding, monkeypatch):
-    monkeypatch.setenv("DD_TRACE_API_VERSION", encoding)
-    caplog.set_level(logging.INFO)
+@pytest.mark.subprocess(parametrize={"DD_TRACE_API_VERSION": SUPPORTED_ENCODINGS})
```

Comment:
You might consider putting this functionality into `parametrize_with_all_encodings` rather than copypasting it throughout this file. Doing so would reduce the cognitive load imposed by this file.